internal/receiver/smartagentreceiver/receiver.go

```diff
@@ -131,7 +131,20 @@ func (r *Receiver) createMonitor(monitorType string, host component.Host) (monit
 
 	monitor = monitorFactory() // monitor is a pointer to a monitor struct
 
-	output := NewOutput(*r.config, r.nextConsumer, host, r.logger)
+	// Make metadata nil if we aren't using built in filtering and then none of
+	// the new filtering logic will apply.
+	metadata, ok := monitors.MonitorMetadatas[monitorType]
+	if !ok || metadata == nil {
+		// This indicates a programming error in not specifying metadata, not
+		// bad user input
+		panic(fmt.Sprintf("could not find monitor metadata of type %s", monitorType))
```

Comment:
The current pattern is to return errors and not panic.  Looking through collector and collector-contrib panicking doesn't seem too common either.

Reply:
converted to a returnen error